### DISCUSSION TITLE
behind-logproperties-and-the-new-telemetry-logging-source-generator/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#398] [Announcements]
# behind-logproperties-and-the-new-telemetry-logging-source-generator/

In this post I look at the new Microsoft.Extensions.Telemetry.Abstractions logging source generator, the generated code, and the [LogProperties] attribute

https://andrewlock.net/behind-logproperties-and-the-new-telemetry-logging-source-generator/

**Comment by joperezr** [comment]
This is awesome! thanks for writing this. This morning I wrote a similar gist but focused on data redaction using the new Microsoft.Extensions.Compliance.Redaction package. Since you mention in the end that up next you plan to tackle data classification, you may find this gist useful. Here is the link: https://gist.github.com/joperezr/f5f022bcb4d0ce8f077e40e1f77239c8#file-redaction-md

**Reply by andrewlock** [reply]
This is great, thanks! And it aligns very well with the post I'm writing, which is always good to see, thanks! 😁

**Comment by alrz** [comment]
Is there any reason to use serilog at this point (assuming you don't need a specific sink)?

**Comment by jdevillard** [comment]
Hello , 

Thanks a lot for this post. 

I'm trying to search some information about the naming convention of state properties using the Logger.  Because, using OpenTelemetry , I've seen the following convention : https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/ which said that attibutes SHOULD be lowercase (and with others rules) . But using ILogger , the samples that define some states attributes are always PascalCase (or mix when using namespace `forecast.TemperatureC`) .  

When using filtering on back end monitoring tools, casing could be important so I would like to find the convention that fit for all :)

Have you some though or guidance on this?

**Comment by jpfed** [comment]
I wonder why the EventId is set to 0 in the new version when it wasn't zero in the old? The fact that it wasn't commented on in the article may mean that it's supposed to be obvious, but I can't think of why.
